pkg/pillar/utils/netutils/ip.go

```diff
@@ -71,19 +74,34 @@ func GetIPAddrCountOnSubnet(subnet net.IPNet) int {
 	return 0
 }
 
+// GetSubnetAddr returns the network (subnet) address of the given IP network.
+// It applies the subnet mask to the IP address to compute the base network address.
+func GetSubnetAddr(ipWithMask *net.IPNet) *net.IPNet {
+	if ipWithMask == nil {
+		return nil
+	}
+	return &net.IPNet{
+		IP:   ipWithMask.IP.Mask(ipWithMask.Mask),
+		Mask: ipWithMask.Mask,
+	}
+}
+
 // GetIPNetwork returns the first IP Address of the subnet(Network Address)
-func GetIPNetwork(subnet net.IPNet) net.IP {
+func GetIPNetwork(subnet *net.IPNet) net.IP {
+	if subnet == nil {
+		return nil
+	}
 	return subnet.IP.Mask(subnet.Mask)
 }
 
 // GetIPBroadcast returns the last IP Address of the subnet(Broadcast Address)
-func GetIPBroadcast(subnet net.IPNet) net.IP {
+func GetIPBroadcast(subnet *net.IPNet) net.IP {
```

Comment:
Should this be renamed to GetIPv4Broadcast() (and only be used in the case of IPv4)?

Reply:
And/or should it check that the argument is IPv4?